src/modm/driver/radio/sx128x_hal.hpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SX128X_HAL_HPP
+#define MODM_SX128X_HAL_HPP
+
+#include "sx128x_definitions.hpp"
+
+#include <modm/architecture/interface/spi_device.hpp>
+#include <modm/processing/resumable.hpp>
+
+namespace modm
+{
+
+/**
+ * SX128X SPI Hardware Abstraction Layer.
```

Comment:
```suggestion
 * SX128X SPI Transport Layer.
```
or Hardware Abstraction Abstraction Layer (HAAL)